tree/treeplayer/src/TTreeReaderArray.cxx

```diff
@@ -564,39 +564,39 @@ void ROOT::Internal::TTreeReaderArrayBase::SetImpl(TBranch* branch, TLeaf* myLea
          if (fSetupStatus == kSetupInternalError)
             fSetupStatus = kSetupMatch;
          if (element->IsA() == TStreamerSTL::Class()){
-            fImpl = new TSTLReader();
+            fImpl = std::make_unique<TSTLReader>();
          }
          else if (element->IsA() == TStreamerObject::Class()){
             //fImpl = new TObjectArrayReader(); // BArray[12]
 
             if (element->GetClass() == TClonesArray::Class()){
-               fImpl = new TClonesReader();
+               fImpl = std::make_unique<TClonesReader>();
             }
             else {
-               fImpl = new TArrayFixedSizeReader(element->GetArrayLength());
+               fImpl = std::make_unique<TArrayFixedSizeReader>(element->GetArrayLength());
             }
          }
          else if (element->IsA() == TStreamerLoop::Class()) {
-            fImpl = new TArrayParameterSizeReader(fTreeReader, branchElement->GetBranchCount()->GetName());
+            fImpl = std::make_unique<TArrayParameterSizeReader>(fTreeReader, branchElement->GetBranchCount()->GetName());
          }
          else if (element->IsA() == TStreamerBasicType::Class()){
             if (branchElement->GetType() == TBranchElement::kSTLMemberNode){
-               fImpl = new TBasicTypeArrayReader();
+               fImpl = std::make_unique<TBasicTypeArrayReader>();
             }
             else if (branchElement->GetType() == TBranchElement::kClonesMemberNode){
-               fImpl = new TBasicTypeClonesReader(element->GetOffset());
+               fImpl = std::make_unique<TBasicTypeClonesReader>(element->GetOffset());
             }
             else {
-               fImpl = new TArrayFixedSizeReader(element->GetArrayLength());
-               ((TObjectArrayReader*)fImpl)->SetBasicTypeSize(((TDataType*)fDict)->Size());
+               fImpl = std::make_unique<TArrayFixedSizeReader>(element->GetArrayLength());
+               ((TObjectArrayReader*)fImpl.get())->SetBasicTypeSize(((TDataType*)fDict)->Size());
```

Comment:
This happens like three times. Any side-effect when using `unique_ptr::get`?

Reply:
That's the correct fix.